src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -29,12 +32,12 @@
     final String Bt_NAME               = "Thaili_Bluetooth";
 
 
-    List<ServiceItem> lastAvailableList = new ArrayList<ServiceItem>();
+    CopyOnWriteArrayList<ServiceItem> lastAvailableList = new CopyOnWriteArrayList<ServiceItem>();
```

Comment:
You really want to make this final as well.  Life is much easier if a variable can never go null. In the case of lastAvailableList, for example, the only place you ever reset it is in AddPeerIfNotDiscovered and you can just call the clear method on the variable.

To be clear, you do NOT have a bug here. The code is correct. The use of final and clear is just meant as a defensive coding technique.